components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/internal/APIHandlerServiceComponent.java

```diff
@@ -353,5 +360,40 @@ protected void unsetMediationSecurityAdminService(MediationSecurityAdminService
 
         ServiceReferenceHolder.getInstance().setMediationSecurityAdminService(null);
     }
+
+    @Reference(
```

Comment:
Can you please elaborate how this method can be used to register multiple implementations?

Reply:
OK, understood how it works. We may not have to necessarily change the approach, but need to think about the following scenarios;
a. Can the component startup order affect the logic?
b. How easy is for the developer to write an extension. Now they need to know certain things about OSGI as well.
We might not have to address these through code, but at least have to provide a good document around doing b .

Reply:
as discussed this could be resolved